jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java

```diff
@@ -504,8 +504,10 @@ private HttpClient buildHttpClient() {
             Header header = new BasicHeader(k, v);
             hdrs.add(header);
         });
-        HttpClient hc = CachingHttpClientBuilder.create().setDefaultHeaders(hdrs).build();
-        return hc;
+        HttpClient hc = HttpOp.createPoolingHttpClientBuilder()
```

Comment:
For future refining, my understanding of `HttpClientBuilder` is that it does exactly what you describe above. IOW, you can call `build()` on it and get a client but keep it around to call `build()` on it later for another same-config client, over and over again.